src/django_components/component.py

```diff
@@ -158,14 +158,32 @@ def get_context_data(self, *args, **kwargs):
     """
 
 
+# Descriptor to pass getting/setting of `template_name` onto `template_file`
+class ComponentTemplateNameDescriptor:
```

Comment:
Because the `template_name` is a class attribute, not an instance property, I had to use descriptors to intercept the access, same as I did in `component_media.py`